### PR TITLE
Feat/handle obj files

### PR DESCRIPTION
**Pull request by mathieurobert1**
Handle the OBJ file:
image

### CODE REVIEW

**Review by theophile-jr — APPROVED** [review]

File: include/Parser/Factory.hpp [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #include <unordered_map>
 #include <libconfig.h++>
 
+#include "Primitives/Mesh.hpp"
 #include "Lights/Point.hpp"
 #include "Lights/Ambient.hpp"
 #include "Primitives/Cone.hpp"
```
Comment (Njord201) [inline include/Parser/Factory.hpp:21]:
```suggestion
#include "Lights/Point.hpp"
#include "Lights/Ambient.hpp"
#include "Primitives/Mesh.hpp"
#include "Primitives/Cone.hpp"
```

File: include/Parser/ParserObj.hpp
Comment (Njord201) [inline include/Parser/ParserObj.hpp:1]: Describe all functions with doxygen

File: include/Parser/ParserObj.hpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+                std::string message;
+        };
+
+    private:
+
+        std::string                                             _filePath;
+        std::shared_ptr<Primitive::Mesh>                        _mesh;
+        std::vector<std::tuple<double, double, double, bool>>   _vertexes;
+        std::vector<std::tuple<int, int, int, bool>>   _faces;
```
Comment (Njord201) [inline include/Parser/ParserObj.hpp:73]: Allign the variable with the other

Comment (Njord201) [reply include/Parser/ParserObj.hpp:73]:
```suggestion
        std::vector<std::tuple<int, int, int, bool>>            _faces;
```

File: include/Primitives/Mesh.hpp [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        Mesh();
+
+        /**
+         * @brief Destroy the Mesh object.
+         *
+         */
+        ~Mesh() = default;
+
+        /**
+         * @brief compute the hit point of a Mesh with a ray
```
Comment (Njord201) [inline include/Primitives/Mesh.hpp:35]:
```suggestion
         * @brief Compute the hit point of a Mesh with a ray.
```

File: include/Primitives/Mesh.hpp [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+namespace Primitive {
+    class Mesh;
+};
+
+class Primitive::Mesh : public Primitive::IPrimitive {
+
+    public:
+
+        /**
+         * @brief Construct a new Mesh object.
```
Comment (Njord201) [inline include/Primitives/Mesh.hpp:23]:
```suggestion
         * @brief Construct a new mesh object.
```

File: include/Primitives/Mesh.hpp [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+         *
+         */
+        Mesh();
+
+        /**
+         * @brief Destroy the Mesh object.
```
Comment (Njord201) [inline include/Primitives/Mesh.hpp:29]:
```suggestion
         * @brief Destroy the mesh object.
```

File: include/Primitives/Mesh.hpp [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        /**
+         * @brief compute the hit point of a Mesh with a ray
+         *
+         * @param ray Vector3D
+         * @return Math::Point3D
+         */
+        Math::Point3D hitPoint(const Raytracer::Ray& ray) override;
+
+        /**
+         * @brief Get the Normal of the object.
```
Comment (Njord201) [inline include/Primitives/Mesh.hpp:43]:
```suggestion
         * @brief Get the normal of the object.
```

File: include/Primitives/Mesh.hpp [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+         *
+         * @param hitPoint to compute the normal
+         * @param ray of the camera
+         * @return Math::Vector3D
+         */
+        Math::Vector3D getNormal(const Math::Vector3D& hitPoint, const Raytracer::Ray& ray) const;
+
+        /**
+         * @brief Set the Material.
```
Comment (Njord201) [inline include/Primitives/Mesh.hpp:52]:
```suggestion
         * @brief Set the material.
```

File: include/Primitives/Mesh.hpp [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+         *
+         * @param material New material to set
+         */
+        void setMaterial(std::shared_ptr<Material::IMaterial> material);
+
+        /**
+         * @brief Get the Material object.
```
Comment (Njord201) [inline include/Primitives/Mesh.hpp:59]:
```suggestion
         * @brief Get the material object.
```

File: include/Primitives/Mesh.hpp [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+        /**
+         * @brief Get the collider box object.
+         *
+         * @return Octree::cubeCollider
+         */
+        Octree::cubeCollider getColliderBox() const;
+
+        /**
+         * @brief Add a Primitive to the Mesh.
```
Comment (Njord201) [inline include/Primitives/Mesh.hpp:73]:
```suggestion
         * @brief Add a primitive to the mesh.
```

File: include/Primitives/Triangle.hpp [hunk 10]
```diff
@@ -35,6 +35,16 @@ namespace Primitive {
...
+             * @param vertex2 second edge of the triangle
+             * @param vertex3 third edge of the triangle
+             */
+            Triangle(const Math::Point3D& vertex1, const Math::Point3D& vertex2, const Math::Point3D& vertex3);
+
+            /**
+             * 
              * @brief Destroy the triangle object.
              *
              */
```
Comment (Njord201) [inline include/Primitives/Triangle.hpp:50]:
```suggestion
            /**
             * @brief Destroy the triangle object.
             *
             */
```

**Review by Njord201 — CHANGES_REQUESTED** [review]

File: include/Primitives/Mesh.hpp [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+        Mesh();
+
+        /**
+         * @brief Destroy the mesh object.
+         *
+         */
+        ~Mesh() = default;
+
+        /**
+         * @brief Compute the hit point of a Mesh with a ray
```
Comment (Njord201) [inline include/Primitives/Mesh.hpp:36]:
```suggestion
         * @brief Compute the hit point of a Mesh with a ray.
```

**Review by Njord201 — CHANGES_REQUESTED** [review]

**Review by Njord201 — APPROVED** [review]
Good job !

**Review by AxelF44 — APPROVED** [review]

**Review by theophile-jr — APPROVED** [review]

**Pull request merged**